text/0000-ember-cli-build-pipeline.md

```diff
@@ -0,0 +1,69 @@
+- Start Date: 2020-01-10
+- Relevant Team(s): Ember CLI
+- RFC PR: https://github.com/emberjs/rfcs/pull/578
+- Tracking:
+
+# Ability to hook into build process without addons
+
+## Summary
+
+This RFC proposes adding the `treeFor` and `treeFor*` hooks to the options passed to the `EmberApp` class.
+These hooks would run during the build process and allow developers to affect the build
+pipelines for each of the trees (app, vendor, public, etc).
+
+## Motivation
+
+Currently, the only supported way to affect the Broccoli tree during build time is by hooking into
+the `treeFor*` APIs provided by addons. Developers that want to use these APIs are forced to develop
+an addon or in-repo addon. This is problematic for a few reasons:
```

Comment:
Whlle this is true, it is not, to my knowledge, documented anywhere and may not even be public api.

Reply:
Thanks for the pointer, @rwjblue, I didn't know about that. Reference for future: https://github.com/ember-cli/ember-cli/blob/v3.18.0/lib/broccoli/ember-app.js#L319-L327

The proposal is to add _additional_ trees, similar to what you do with addons, rather than having to start from scratch with a completely custom trees.